pkg/controller/testutils/testutils.go

```diff
@@ -32,3 +37,16 @@ func GetReconcileMetricCount(t *testing.T, controllerName, label string) float64
 	beforeCount := metricProto.GetCounter().GetValue()
 	return beforeCount
 }
+
+func StartTestingEnv() (*rest.Config, *envtest.Environment, error) {
+	env = &envtest.Environment{}
```

Comment:
slight race condition of still using the global variable.

Two tests can be in startTestingEnv at the same time and overwrite eachother's env.